presidio-api/cmd/presidio-api/api/analyze/analyze_test.go

```diff
@@ -88,3 +88,20 @@ func TestLanguageCode(t *testing.T) {
 	Analyze(context.Background(), api, analyzeAPIRequest, project)
 	assert.Equal(t, "langtest", analyzeAPIRequest.AnalyzeTemplate.Language)
 }
+
+func TestAllFields(t *testing.T) {
+
+	api := setupMockServices()
+
+	project := "tests"
+	analyzeAPIRequest := &types.AnalyzeApiRequest{
+		Text:              "My number is (555) 253-0000 and email johnsnow@foo.com",
+		AnalyzeTemplateId: "test",
```

Comment:
The template Id field is redundant as you already pass the template itself.
will be a good idea to also verify this in the analyze request itself (different pr)